README.md

```diff
@@ -43,30 +43,61 @@ $ git clone https://github.com/iterative/example-gto.git
 $ cd example-gto
 ```
 
-### Versioning
+This guide will walk you through the Lifecycle of an artifact. To execute each
+step, you'll need to create a Git tag of a special format, which will trigger CI
+and trigger the job you intend to run:
 
-To register a new artifact or a new version, use `gto register`. This is usually
-done to mark significant changes to the artifact (such as a release or a
-deprecation).
+1. Registering artifact - marking the very beginning of the artifact history in
+   the Git repository.
+2. Publishing a version - marking an important change in the artifact.
+3. Assigning a stage to a version - marking the readiness to be consumed by a
+   downstream system.
+4. Unassigning a stage - removing an "artifact is ready for this stage" mark.
+5. Deprecating a version - marking a version as the one that should not be used
+   any longer.
+6. Deregistering an artifact - completing the lifecycle of an artifact and
+   marking it as an outdated one.
+
+### Registering an artifact
```

Comment:
This is confusing, coz in Studio model registry, you see models that you annotated even if you have not registered them. Should the `Add a model` flow in Studio also call `register` and create the Git tag to register the model? Is it possible to combine these commands somehow, so that the user does not have to use so many commands (register, publish, annotate)?

Reply:
Good question. I think it should not. We can explain that artifact is considered to be `registered` if there is any version (that wasn't deprecated/unpublished).

Also I recently understood that what Studio shows is the same you should get if you run GTO on a remote repo, e.g. `$ gto show https://github.com/iterative/model-registry`. GTO doesn't process remote repos yet, but we'll add it soon #220

Reply:
The difference from Studio still is that when you add a model from Studio, we don't call the GTO `register` api and create the model tag without versions. I think we should make this consistent with GTO and also call `register`. Also, IMO having to run 3 commands (register, publish, annotate) for getting your first version up is too much. Can `register` internally call `publish` to register the first version? And can it also accept model description etc and call `annotate` internally?

Reply:
I agree with the reasoning, but I'm afraid combining `register` and `annotate` in a single command is also confusing. Consider, that `annotate` edits `artifacts.yaml` and don't create a commit with this change, while `register` creates a git tag for an already existing commit. If a single command both changes files in the current index and creates a git tag, that can be quite confusing IMO.

Reply:
Thanks @tapadipti for pointing out this! I would expect that during the [`Add Model`](https://dvc.org/doc/studio/user-guide/model-registry/add-a-model) scenario Studio calls 2 commands 
- `gto register`   -> create a tag 
- `gto annotate` 

Not sure about that a separate command is necessary, but I would upvote for consistent behavior!

Reply:
We at least need a separate `gto register` flow in Studio. May skip separate `gto annotate` flow, if you ask me.

Reply:
@aguschin Sorry for the late reply. Not sure I got you. Do you mean that **in addition to the current Add a model flow**, we need the `gto register` flow in Studio? Or that we don't need `gto annotate` flow? If the latter, then how do we save the model description / labels / etc.?

I agree with @mnrozhkov - I think we need to call both `gto register` and `gto annotate` for consistency. @iterative/studio-backend wdyt?

Reply:
@tapadipti, you're right - we have `gto register` flow already:
<img width="190" alt="image" src="https://user-images.githubusercontent.com/6797716/185275330-043030c3-b4c7-4edc-ae39-3a494694e5d2.png">
Sorry for the confusion :)

Reply:
Yes, we can call both annotate and register from Studio when adding model. If we do this, are we to  assume that any model will have at least 1 version registered? 
Should we ignore the models without any version registered from now on?

I am trying to understand the reason to register the model when user may want to just add or annotate a model in artifacts.yaml file at the time.

Reply:
> Should we ignore the models without any version registered from now on?

don't think so.

As I understand this the reason is to simplify user workflow. There are cases when a user would want to annotate and register a version at the same time, so it would be convenient to provide an option to do that. At the same time, someone may want to annotate without registering, that's also fine.

Reply:
@aguschin Questions:
- In terms of displaying the model in the Studio model registry, how should we distinguish between models that are only annotated and models that are annotated and registered? Should we treat them the same and display both these types of models in the MR?
- In terms of the `Add a model` flow in Studio, should we ask the users if they want to only annotate or annotate&register? I think this will introduce unnecessary confusion to the users. Can we agree on one flow?

Reply:
> how should we distinguish between models that are only annotated and models that are annotated and registered

Please check out [this issue](https://github.com/iterative/gto/issues/226). Maybe this is a solution.

on a side note, a similar thing in GTO was proposed by @jorgeorpinel a while ago https://github.com/iterative/gto/issues/145

Reply:
@tapadipti, do we have an issue for this already or should we create one?

Reply:
No we don't. I'm reading https://github.com/iterative/gto/issues/226 and it looks like we should conclude that discussion and then create required tickets in Studio. Has there been any discussion on that?

Reply:
No, nothing as far as I know.